lib/python/bind_slice_methods.h

```diff
@@ -324,12 +333,26 @@ void bind_slice_methods(pybind11::class_<T, Ignored...> &c) {
           slicer<T>::template set<std::tuple<Dim, py::slice>>(
               self, {self.dim(), index}, data);
         });
-  c.def("__setitem__", &slicer<T>::template set<std::tuple<Dim, scipp::index>>);
-  c.def("__setitem__", &slicer<T>::template set<std::tuple<Dim, py::slice>>);
-  c.def("__setitem__", &slicer<T>::template set<py::ellipsis>);
-  c.def("__getitem__", [](T &self, const std::vector<scipp::index> &indices) {
-    expect_implicit_dimension(self.dims());
-    return slice_by_list(self, {self.dim(), indices});
+  c.def("__setitem__", [](T &self, std::tuple<std::string, scipp::index> index,
+                          const py::object &data) {
+    slicer<T>::template set(self, to_dim_type(std::move(index)), data);
   });
-  c.def("__getitem__", slice_by_list<T>);
+  c.def("__setitem__", [](T &self, std::tuple<std::string, py::slice> index,
+                          const py::object &data) {
+    slicer<T>::template set(self, to_dim_type(std::move(index)), data);
+  });
+  c.def("__setitem__", &slicer<T>::template set<py::ellipsis>);
+  c.def(
+      "__getitem__",
+      [](T &self, const std::vector<scipp::index> &indices) {
+        expect_implicit_dimension(self.dims());
+        return slice_by_list(self, {self.dim(), indices});
+      },
+      py::call_guard<py::gil_scoped_release>());
+  c.def(
+      "__getitem__",
+      [](T &self, std::tuple<std::string, std::vector<scipp::index>> indices) {
+        return slice_by_list<T>(self, to_dim_type(std::move(indices)));
```

Comment:
In all of the above, wouldn't it be easier to change the functions such as `getitem` and `slice_by_list` to do the conversion to `Dim`, instead of doing it in every call?

Reply:
No, because of these overloads which would have to convert `Dim -> string -> Dim`
https://github.com/scipp/scipp/blob/7229fedd536dc163e3e4c2a8576885886510b496/lib/python/bind_slice_methods.h#L263-L271
https://github.com/scipp/scipp/blob/7229fedd536dc163e3e4c2a8576885886510b496/lib/python/bind_slice_methods.h#L345-L351